### PR TITLE
Feature/1658 patch subs v2

### PR DESCRIPTION
**Pull request by crbrox**
@fgalan 
@kzangeli

### CODE REVIEW

File: src/app/contextBroker/contextBroker.cpp [hunk 1]
```diff
@@ -195,6 +195,7 @@
 #include "serviceRoutinesV2/getSubscription.h"
```
Comment (fgalan) [inline src/app/contextBroker/contextBroker.cpp]: CNR entry missing, e.g:

Add: PATCH /v2/subscriptions/{subId} operation (Issue #1658)

File: src/lib/jsonParseV2/parseSubscription.cpp [hunk 2]
```diff
@@ -175,9 +195,8 @@ static std::string parseSubject(ConnectionInfo* ciP, SubscribeContextRequest* sc
     r = parseEntitiesVector(ciP, &scrP->entityIdVector, subject["entities"]);
     if (r != "") {
       return r;
-    }
+    };
```
Comment (fgalan) [inline src/lib/jsonParseV2/parseSubscription.cpp]: I understand that ";" is not needed after this "}"...

File: src/lib/mongoBackend/mongoUpdateContextSubscription.cpp [hunk 3]
```diff
@@ -107,14 +111,84 @@ HttpStatusCode mongoUpdateContextSubscription
...
+  {
+    // Reference
+    std::string ref;
+    if (!requestP->reference.isEmpty())
+    {
+      ref = requestP->reference.get();
+    }
+    else
+    {
+      ref = getStringField(sub, CSUB_REFERENCE);
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoUpdateContextSubscription.cpp]: In a recent PR, we instroduced reference URL checking. Just to ensure that this PR keep that checking (maybe it is done at parsing stage?).

File: src/app/contextBroker/contextBroker.cpp (hunk 1 above)
Comment (crbrox) [inline src/app/contextBroker/contextBroker.cpp]: Fixed 14e0b5c

File: src/lib/mongoBackend/mongoUpdateContextSubscription.cpp [hunk 4]
```diff
@@ -126,7 +200,7 @@ HttpStatusCode mongoUpdateContextSubscription
   else
   {
     newSub.append(CSUB_EXPIRATION, expiration);
-    LM_T(LmtMongo, ("New subscription expiration: %l", expiration));
+    LM_T(LmtMongo, ("New subscription expiration: %" PRId64, expiration));
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoUpdateContextSubscription.cpp]: What is "PRId64" ?

File: src/lib/ngsi10/UpdateContextSubscriptionRequest.h [hunk 5]
```diff
@@ -35,28 +35,22 @@
...
 
 
 /* ****************************************************************************
 *
 * UpdateContextSubscriptionRequest - 
 */
-typedef struct UpdateContextSubscriptionRequest
+struct UpdateContextSubscriptionRequest : public SubscribeContextRequest
 {
-  Duration                       duration;               // Optional
```
Comment (fgalan) [inline src/lib/ngsi10/UpdateContextSubscriptionRequest.h]: I understand that all the removed fields (duration, restriction, notifyConditionVector, throttling, restrictions) are got from the base clase. Is that correct?

File: src/lib/jsonParseV2/parseSubscription.cpp (hunk 2 above)
Comment (crbrox) [inline src/lib/jsonParseV2/parseSubscription.cpp]: Fixed b55a1cb

File: test/functionalTest/cases/1316_create_subscription/post_subscriptions.test [hunk 6]
```diff
@@ -186,7 +186,8 @@ Date: REGEX(.*)
             "humidity",
             "temperature"
         ],
-        "callback": "http://localhost:1234"
+        "callback": "http://localhost:1234",
+        "throttling": 5
```
Comment (fgalan) [inline test/functionalTest/cases/1316_create_subscription/post_subscriptions.test]: There was a bug (now fixed) precluding throttling to appear in the GET subscriptions information?

File: src/lib/ngsi10/UpdateContextSubscriptionRequest.h (hunk 5 above)
Comment (crbrox) [inline src/lib/ngsi10/UpdateContextSubscriptionRequest.h]: Yes, it is.

File: src/lib/mongoBackend/mongoUpdateContextSubscription.cpp (hunk 4 above)
Comment (crbrox) [inline src/lib/mongoBackend/mongoUpdateContextSubscription.cpp]: a "constant" (macro for a string literal) for the right format specifier for int64_t

File: test/functionalTest/cases/1316_create_subscription/post_subscriptions.test (hunk 6 above)
Comment (crbrox) [inline test/functionalTest/cases/1316_create_subscription/post_subscriptions.test]: In the POST, I'd say

Comment (fgalan) [inline test/functionalTest/cases/1316_create_subscription/post_subscriptions.test]: Ok. Good cach!

NTC

File: src/lib/mongoBackend/mongoUpdateContextSubscription.cpp (hunk 3 above)
Comment (crbrox) [inline src/lib/mongoBackend/mongoUpdateContextSubscription.cpp]: Fixed 22fd8b1

File: src/lib/mongoBackend/mongoUpdateContextSubscription.cpp (hunk 4 above)
Comment (fgalan) [inline src/lib/mongoBackend/mongoUpdateContextSubscription.cpp]: Nunca te acostarás sin saber una cosa más... :)

NTC

File: src/lib/ngsi10/UpdateContextSubscriptionRequest.h (hunk 5 above)
Comment (fgalan) [inline src/lib/ngsi10/UpdateContextSubscriptionRequest.h]: NTC

**Comment by fgalan** [comment]
LGTM

**Comment by kzangeli** [comment]
LGTM (still needs a merge with develop)

**Pull request merged**